In [1]:
import pandas as pd
import numpy as np

Thanks to Kriti Mahajan. This table was developed with study type features extracted from the meta data.
Code found here: https://colab.research.google.com/drive/1lay-yWJ5ohN94lQLHYIxtWMuDWtCRGy0#scrollTo=N-IhAeEkcxDe

In [2]:
mf = pd.read_csv("metadata_features.csv")
mf

Unnamed: 0  paper_id  \
0               0  zjufx4fo   
1               1  ymceytj3   
2               2  wzj2glte   
3               3  2sfqsfm1   
4               4  i0zym7iq   
...           ...       ...   
56640       56640  rn4gm3sc   
56641       56641  l5oecoot   
56642       56642  i4acg2ex   
56643       56643  rve0w3dg   
56644       56644  ut7dlyh8   

                                                   title  \
0      Sequence requirements for RNA strand transfer ...   
1      Crystal structure of murine sCEACAM1a[1,4]: a ...   
2      Synthesis of a novel hepatitis C virus protein...   
3      Structure of coronavirus main proteinase revea...   
4      Discontinuous and non-discontinuous subgenomic...   
...                                                  ...   
56640  Structure-Based Stabilization of Non-native Pr...   
56641          Immunomodulators as an antimicrobial tool   
56642  Discourses of disease: Representations of tube...   
56643                           Equine Infectious Anemia   
56644  A duplex real-time reverse transcriptase polym...   

                                                abstract publish_time  \
0      Nidovirus subgenomic mRNAs contain a leader se...   2001-12-17   
1      CEACAM1 is a member of the carcinoembryonic an...   2002-05-01   
2      Hepatitis C virus (HCV) is an important human ...   2001-07-16   
3      The key enzyme in coronavirus polyprotein proc...   2002-07-01   
4      Arteri-, corona-, toro- and roniviruses are ev...   2002-12-01   
...                                                  ...          ...   
56640  [Image: see text] Structure-based stabilizatio...   2020-02-27   
56641  The spectrum of infectious diseases has shifte...   2006-10-31   
56642  Abstract This paper critically examines the wa...   2008-02-29   
56643  This article describes clinical signs and tran...   1993-08-31   
56644  Abstract A duplex TaqMan real-time reverse tra...   2009-10-31   

                                                 authors  \
0      Pasternak, Alexander O.; van den Born, Erwin; ...   
1      Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...   
2      Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...   
3      Anand, Kanchan; Palm, Gottfried J.; Mesters, J...   
4      van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...   
...                                                  ...   
56640  Lin, Shan-Meng; Lin, Shih-Chao; Hsu, Jia-Ning;...   
56641           Pirofski, Liise-anne; Casadevall, Arturo   
56642  Lawrence, Jody; Kearns, Robin A.; Park, Julie;...   
56643                             Sellon, Debra Clabough   
56644  Wang, Heng; Nattanmai, Seela; Kramer, Laura D....   

                                                     url Statistical Methods  \
0      http://europepmc.org/articles/pmc125340?pdf=re...                  []   
1      http://europepmc.org/articles/pmc125375?pdf=re...                  []   
2      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...                  []   
3      http://europepmc.org/articles/pmc126080?pdf=re...           ['model']   
4      http://europepmc.org/articles/pmc136939?pdf=re...                  []   
...                                                  ...                 ...   
56640  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                  []   
56641          https://doi.org/10.1016/j.mib.2006.08.004           [' tool']   
56642    https://doi.org/10.1016/j.socscimed.2007.10.015                  []   
56643      https://doi.org/10.1016/s0749-0739(17)30399-1                  []   
56644  https://doi.org/10.1016/j.diagmicrobio.2009.07...                  []   

                              statistical_method_details  \
0                                                    NaN   
1                                                    NaN   
2                                                    NaN   
3      Molecular modelling and mutagenesis data impli...   
4                                                    N

In [3]:
list(mf)

['Unnamed: 0',
 'paper_id',
 'title',
 'abstract',
 'publish_time',
 'authors',
 'url',
 'Statistical Methods',
 'statistical_method_details',
 'Study Type',
 'study_type_details',
 'results',
 'coronavirus',
 'fatality',
 'sample_size',
 'paper_focus',
 'variables_controlled_for']

In [4]:
mf['Study Type'].value_counts()

[]                                                                                       26228
['protein']                                                                               6699
['clinical']                                                                              4373
['review']                                                                                2533
['genom']                                                                                 1251
                                                                                         ...  
['review', 'cross section', 'cross sectional', 'clinical']                                   1
['retrospective study', 'prospective study', 'clinical']                                     1
['genom', 'genetic', 'life cycle', 'clinical']                                               1
['prospective cohort', 'prospective cohort', 'prospective observational', 'clinical']        1
['protein', 'mutagenesis', 'genetic', 'experimenta

Initially, will work with this and see F1 score. Will then create own features by traditional feature engineering and reverse engineering to get cleanest features set possible. Caveat with this preliminary feature set is that when [] is taken out, 46% of the metadata is not accounted for (no study type key words). Can remedy this through changing n-gram feature-set to be more specific and relevant/feature engineering.

## Features Preprocessing

In [5]:
features = mf.drop(columns = ['Unnamed: 0',
 'title',
 'abstract',
 'publish_time',
 'authors',
 'url',
 'Statistical Methods',
 'statistical_method_details',
 'study_type_details',
 'results',
 'coronavirus',
 'fatality',
 'sample_size',
 'paper_focus',
 'variables_controlled_for'])
features

paper_id                  Study Type
0      zjufx4fo    ['genom', 'mutagenesis']
1      ymceytj3                 ['protein']
2      wzj2glte        ['protein', 'genom']
3      2sfqsfm1  ['protein', 'mutagenesis']
4      i0zym7iq                   ['genom']
...         ...                         ...
56640  rn4gm3sc                 ['protein']
56641  l5oecoot                          []
56642  i4acg2ex                          []
56643  rve0w3dg      ['review', 'clinical']
56644  ut7dlyh8      ['protein', 'genetic']

[56645 rows x 2 columns]

In [6]:
features.isnull().sum()

paper_id      0
Study Type    0
dtype: int64

In [54]:
#Get rid of rows with empty lists (STRING TYPE) since void of study type features.
# test2 =  features[features['Study Type'].astype(bool)]
# test2
to_drop = ['[]']
features = features[~features['Study Type'].isin(to_drop)]
features

paper_id                  Study Type
0      zjufx4fo    ['genom', 'mutagenesis']
1      ymceytj3                 ['protein']
2      wzj2glte        ['protein', 'genom']
3      2sfqsfm1  ['protein', 'mutagenesis']
4      i0zym7iq                   ['genom']
...         ...                         ...
56638  jna496x7                 ['protein']
56639  4jg4t8po                 ['protein']
56640  rn4gm3sc                 ['protein']
56643  rve0w3dg      ['review', 'clinical']
56644  ut7dlyh8      ['protein', 'genetic']

[30417 rows x 2 columns]

## Joining feature set to label set (Kaggle for now)

In order to attach study type features to labels, need to use CoronaWhy annotations dataset that has cord_uid that matches with paper_id from above as well as sha that matches with id in label dataset.

In [44]:
get_cord_uid = pd.read_csv("Study_Design_Annotation_Snapshot_4_12_20.csv")
labels = pd.read_csv("design.csv") 

In [47]:
list(mf_cordid)

['paper_id',
 'Study Type',
 'Assignee',
 'cord_uid',
 'sha',
 'title',
 'url',
 'in silico',
 'in vitro',
 'in vivo',
 'Systematic reivew or meta-analysis of RCTs',
 'RCT',
 'Non-randomized controlled trial',
 'Comparative study with a concurrent or historical comparator (cohort studies, case-control studies, 2 or more single-arm studies, or interrupted time series with a control group)',
 'Descriptive study without a comparator (single-arm pre-post studies, case series, case reports)',
 'Systematic review or meta-analysis of studies other than RCTs',
 'Other ',
 'General Notes']

In [55]:
feat_cordid = features.merge(get_cord_uid, how='inner', left_on='paper_id', right_on='cord_uid')
feat_cordid = feat_cordid.drop(columns = ['Assignee',
 'title',
 'url',
 'in silico',
 'in vitro',
 'in vivo',
 'Systematic reivew or meta-analysis of RCTs',
 'RCT',
 'Non-randomized controlled trial',
 'Comparative study with a concurrent or historical comparator (cohort studies, case-control studies, 2 or more single-arm studies, or interrupted time series with a control group)',
 'Descriptive study without a comparator (single-arm pre-post studies, case series, case reports)',
 'Systematic review or meta-analysis of studies other than RCTs',
 'Other ',
 'General Notes'] )
feat_cordid

paper_id                   Study Type  cord_uid  \
0    lwla5ugt  ['genom', 'bioinformatics']  lwla5ugt   
1    gdsfkw1b        ['protein', 'review']  gdsfkw1b   
2    e1iaiwc6                   ['survey']  e1iaiwc6   
3    o088rxb5                 ['clinical']  o088rxb5   
4    icp9d998                 ['clinical']  icp9d998   
..        ...                          ...       ...   
595  f7i4lq2w      ['genetic', 'clinical']  f7i4lq2w   
596  hwt1foyu                  ['protein']  hwt1foyu   
597  c9zya2t7                  ['protein']  c9zya2t7   
598  437zx0xr                   ['survey']  437zx0xr   
599  0y5jaf1m                  ['protein']  0y5jaf1m   

                                          sha  
0    8970ef10049816050cb1bc5fbf600865a5996ad4  
1    8d7400a2b387820cd391d7df8194642e50402a0c  
2    eddc547cbba693715e4fd55e6a946e8ec7d96bc2  
3    9f63ebfaab049c0968d72166761217b3aaa1fe00  
4    8c640872048570b0a671ec08c83ef6cbebcdd78e  
..                                        ...  
595  450921cad3744aa651260fdd482038f546992999  
596  ce0f694c2a6f8568c566118e14bef4c1898957f6  
597  2b7cc2e066ab4f8252246b530424d878b260b108  
598  9da58647ca34a42efcf47cec929b7163a808a65d  
599  60a342d1b18b2059aa722f46cc9f8b9770b6919c  

[600 rows x 4 columns]

In [57]:
labels_merge = feat_cordid.merge(labels, how='inner', left_on='sha', right_on='id')
labels_merge

paper_id                                      Study Type  cord_uid  \
0  7n75m669                                     ['genetic']  7n75m669   
1  ie3msqwt                                     ['genetic']  ie3msqwt   
2  u1m7yr8f  ['questionnaire', 'survey', 'cross-sectional']  u1m7yr8f   
3  a27prjq9                                      ['review']  a27prjq9   
4  9vrny92m                                      ['review']  9vrny92m   

                                        sha  label  \
0  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d      8   
1  5b15c62ee761d6eefeb7d2ef023f804bfdc41edc      9   
2  eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad      7   
3  4bc6ead9368ec84dba7c56b78624f681293e568b      7   
4  d7d62e768f4a8eb40ba522eca28e173e5d53cbbc      4   

                                         id  
0  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d  
1  5b15c62ee761d6eefeb7d2ef023f804bfdc41edc  
2  eac85c85c3a2ac1d3fe53e1e3b3fee98f52527ad  
3  4bc6ead9368ec84dba7c56b78624f681293e568b  
4  d7d62e768f4a8eb40ba522eca28e173e5d53cbbc

In [58]:
list(labels_merge)

['paper_id', 'Study Type', 'cord_uid', 'sha', 'label', 'id']

In [59]:
final = labels_merge.drop(columns = ['cord_uid', 'sha', 'id'])
final

paper_id                                      Study Type  label
0  7n75m669                                     ['genetic']      8
1  ie3msqwt                                     ['genetic']      9
2  u1m7yr8f  ['questionnaire', 'survey', 'cross-sectional']      7
3  a27prjq9                                      ['review']      7
4  9vrny92m                                      ['review']      4

Only 5 results. This is a sad. It is due to new dataset being larger (10,000 more papers). So there are going to be more dissimilar paper_ids.  In order to limit dataset reduction, need to see if featuers dataframe can be remade with longer(sha?) paper_id in order to match with labels dataframe directly.

If this does not work out, will then need to synthesize labels. So to summarize:
1. Max out matches between feature set and label set
2. Feature engineer via revising ngram list/reverse engineer
3. Synthetic labels for mutlilabels annotations

### Trying direct merge with mf and kaggle label set via sha.

In [69]:
mf2 = pd.read_csv('metadata_features-2.csv')
mf2

Unnamed: 0  paper_id  \
0               0  zjufx4fo   
1               1  ymceytj3   
2               2  wzj2glte   
3               3  2sfqsfm1   
4               4  i0zym7iq   
...           ...       ...   
59125       59125  pwq5gir9   
59126       59126  3racv2qe   
59127       59127  jngto9kv   
59128       59128  td0776wj   
59129       59129  jxbr6hbk   

                                                     sha  \
0               b2897e1277f56641193a6db73825f707eed3e4c9   
1               e3d0d482ebd9a8ba81c254cc433f314142e72174   
2               00b1d99e70f779eb4ede50059db469c65e8c1469   
3               cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   
4               dde02f11923815e6a16a31dd6298c46b109c5dfa   
...                                                  ...   
59125           20f185b11fa23da8938efb5c839c6c078c66f1e9   
59126           e84532333db8f219f20ae0a8fe18130c0d010b04   
59127           31105078a2953217223699d09c6a80d0f5edfdf6   
59128           b1f977d58a0eba25acdee5912ef71c372b0632de   
59129  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   

                                                   title  \
0      Sequence requirements for RNA strand transfer ...   
1      Crystal structure of murine sCEACAM1a[1,4]: a ...   
2      Synthesis of a novel hepatitis C virus protein...   
3      Structure of coronavirus main proteinase revea...   
4      Discontinuous and non-discontinuous subgenomic...   
...                                                  ...   
59125  Cutting-Edge Infectious Disease Diagnostics wi...   
59126  Intranasal administration of oxytocin: Behavio...   
59127  Infecciones virales graves en pacientes inmuno...   
59128  18 Omics, Bioinformatics, and Infectious Disea...   
59129  Identifying spatio-temporal dynamics of Ebola ...   

                                                abstract publish_time  \
0      Nidovirus subgenomic mRNAs contain a leader se...   2001-12-17   
1      CEACAM1 is a member of the carcinoembryonic an...   2002-05-01   
2      Hepatitis C virus (HCV) is an important human ...   2001-07-16   
3      The key enzyme in coronavirus polyprotein proc...   2002-07-01   
4      Arteri-, corona-, toro- and roniviruses are ev...   2002-12-01   
...                                                  ...          ...   
59125  Three recent Science articles (Chen et al., 20...   2018-06-13   
59126  Abstract The intranasal (IN-) administration o...   2013-09-30   
59127  Resumen Los virus tienen un papel importante d...   2011-04-30   
59128  Bioinformatics and the ‘omics’ studies are rap...   2011-12-31   
59129  Containing the recent West African outbreak of...   2017-11-29   

                                                 authors  \
0      Pasternak, Alexander O.; van den Born, Erwin; ...   
1      Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu...   
2      Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; ...   
3      Anand, Kanchan; Palm, Gottfried J.; Mesters, J...   
4      van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M...   
...                                                  ...   
59125                                      Chiu, Charles   
59126                   Veening, Jan G.; Olivier, Berend   
59127  Díaz, A.; Zaragoza, R.; Granada, R.; Salavert, M.   
59128       Paszkiewicz, Konrad H.; Giezen, Mark van der   
59129             Gustafson, Kyle B.; Proctor, Joshua L.   

                                                     url  \
0      http://europepmc.org/articles/pmc125340?pdf=re...   
1      http://europepmc.org/articles/pmc125375?pdf=re...   
2      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...   
3      http://europepmc.org/articles/pmc126080?pdf=re...   
4      http://europepmc.org/articles/pmc136939?pdf=re...   
...                                                  ...   
59125         https://doi.org/10.1016/j.chom.2018.05.016   
59126    https://doi.org/10.1016/j.neubiorev.2013.04.012   
59127        https://doi.org/10.1016/j.medin.2011.01.0

In [72]:
list(mf2)

['Unnamed: 0',
 'paper_id',
 'sha',
 'title',
 'abstract',
 'publish_time',
 'authors',
 'url',
 'Statistical Methods',
 'statistical_method_details',
 'Study Type',
 'study_type_details',
 'results',
 'coronavirus',
 'fatality',
 'sample_size',
 'paper_focus',
 'variables_controlled_for']

In [73]:
features2 = mf2.drop(columns = ['Unnamed: 0',
 'title',
 'abstract',
 'publish_time',
 'authors',
 'url',
 'Statistical Methods',
 'statistical_method_details',
 'study_type_details',
 'results',
 'coronavirus',
 'fatality',
 'sample_size',
 'paper_focus',
 'variables_controlled_for'])
features2

paper_id                                                sha  \
0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   
1      ymceytj3           e3d0d482ebd9a8ba81c254cc433f314142e72174   
2      wzj2glte           00b1d99e70f779eb4ede50059db469c65e8c1469   
3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   
4      i0zym7iq           dde02f11923815e6a16a31dd6298c46b109c5dfa   
...         ...                                                ...   
59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   
59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   
59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   
59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   
59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   

               Study Type  
0         ['mutagenesis']  
1                      []  
2                      []  
3         ['mutagenesis']  
4                      []  
...                   ...  
59125        ['clinical']  
59126          ['review']  
59127          ['review']  
59128  ['bioinformatics']  
59129         ['genetic']  

[59130 rows x 3 columns]

In [74]:
to_drop = ['[]']
features2 = features2[~features2['Study Type'].isin(to_drop)]
features2

paper_id                                                sha  \
0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   
3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   
6      gi6uaa83           8ae137c8da1607b3a8e4c946c07ca8bda67f88ac   
7      0m32ecnu           23bc55d6f63fab18b02004483888db2b6a0bfa48   
10     fy4w7xz8           0104f6ceccf92ae8567a0102f89cbb976969a774   
...         ...                                                ...   
59125  pwq5gir9           20f185b11fa23da8938efb5c839c6c078c66f1e9   
59126  3racv2qe           e84532333db8f219f20ae0a8fe18130c0d010b04   
59127  jngto9kv           31105078a2953217223699d09c6a80d0f5edfdf6   
59128  td0776wj           b1f977d58a0eba25acdee5912ef71c372b0632de   
59129  jxbr6hbk  4eb170db0dd666fc669649d0e6b15e513b6598d6; 9403...   

                     Study Type  
0               ['mutagenesis']  
3               ['mutagenesis']  
6      ['genetic', 'histories']  
7                   ['genetic']  
10                  ['genetic']  
...                         ...  
59125              ['clinical']  
59126                ['review']  
59127                ['review']  
59128        ['bioinformatics']  
59129               ['genetic']  

[21864 rows x 3 columns]

In [75]:
labels_merge2 = features2.merge(labels, how='inner', left_on='sha', right_on='id')
labels_merge2

paper_id                                       sha  \
0     47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   
1     ub9dtgup  78326020e9f7cd75415a3a192729ccf0c0930281   
2     8f0pn321                                       NaN   
3     xavtxl4m                                       NaN   
4     ciuhyhb8                                       NaN   
...        ...                                       ...   
3660  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   
3661  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   
3662  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   
3663  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   
3664  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   

                                          Study Type  label  \
0                                         ['review']      1   
1                       ['genetic', 'questionnaire']      8   
2                                         ['survey']      2   
3                                         ['survey']      2   
4                                       ['clinical']      2   
...                                              ...    ...   
3660                                    ['clinical']      2   
3661                                    ['clinical']     10   
3662                                ['experimental']      2   
3663  ['questionnaire', 'survey', 'cross-sectional']      7   
3664                                  ['case study']      9   

                                            id  
0     6a3fa8ed278df0d05c5e009521de11c72308f60b  
1     78326020e9f7cd75415a3a192729ccf0c0930281  
2                                          NaN  
3                                          NaN  
4                                          NaN  
...                                        ...  
3660  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e  
3661  eb720970f362e4416b73393f230c8606c837c6dc  
3662  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  
3663  95d9bac431797e2cf8b7294a648ceec3bd50c84b  
3664  76edb6d0bd7c47765d3a89032aa610f410506988  

[3665 rows x 5 columns]

Check for duplicates. (There are probably duplicates because sha is not unique identifier but indicator of change)

In [83]:
labels_merge2.sha.nunique()

429

In [85]:
labels_merge2.paper_id.nunique()

3665

There are 429 shared papers between sha and id. However, what produces the 3665 results for labels_merge2 is the paper_id column that there are  sha dupilicates. So if we remove duplicates, will get 429 results for modeling.

In [88]:
labels_merge2 = labels_merge2.drop_duplicates(subset=['sha'])

final = pd.merge(labels_merge2, labels, how='inner', left_on='sha', right_on='id')
final

paper_id                                       sha  \
0    47ema2dq  6a3fa8ed278df0d05c5e009521de11c72308f60b   
1    ub9dtgup  78326020e9f7cd75415a3a192729ccf0c0930281   
2    8f0pn321                                       NaN   
3    9trzqzep  eb4b8f43f68db5e493984b7e4e56d9f804d7e885   
4    7n75m669  5e9ccecdd40825f5c30da3a900fc5cf1b063b47d   
..        ...                                       ...   
425  l4649du3  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e   
426  3vpoxfnv  eb720970f362e4416b73393f230c8606c837c6dc   
427  p8vhi6d9  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb   
428  a6i73v0o  95d9bac431797e2cf8b7294a648ceec3bd50c84b   
429  nbbcacy4  76edb6d0bd7c47765d3a89032aa610f410506988   

                                         Study Type  label_x  \
0                                        ['review']        1   
1                      ['genetic', 'questionnaire']        8   
2                                        ['survey']        2   
3              ['questionnaire', 'cross-sectional']        9   
4                                       ['genetic']        8   
..                                              ...      ...   
425                                    ['clinical']        2   
426                                    ['clinical']       10   
427                                ['experimental']        2   
428  ['questionnaire', 'survey', 'cross-sectional']        7   
429                                  ['case study']        9   

                                         id_x  label_y  \
0    6a3fa8ed278df0d05c5e009521de11c72308f60b        1   
1    78326020e9f7cd75415a3a192729ccf0c0930281        8   
2                                         NaN        2   
3    eb4b8f43f68db5e493984b7e4e56d9f804d7e885        9   
4    5e9ccecdd40825f5c30da3a900fc5cf1b063b47d        8   
..                                        ...      ...   
425  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e        2   
426  eb720970f362e4416b73393f230c8606c837c6dc       10   
427  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb        2   
428  95d9bac431797e2cf8b7294a648ceec3bd50c84b        7   
429  76edb6d0bd7c47765d3a89032aa610f410506988        9   

                                         id_y  
0    6a3fa8ed278df0d05c5e009521de11c72308f60b  
1    78326020e9f7cd75415a3a192729ccf0c0930281  
2                                         NaN  
3    eb4b8f43f68db5e493984b7e4e56d9f804d7e885  
4    5e9ccecdd40825f5c30da3a900fc5cf1b063b47d  
..                                        ...  
425  bb3bf36ce2af876dcbf80b6cf3598e4421c98b3e  
426  eb720970f362e4416b73393f230c8606c837c6dc  
427  f778c6317cdc3e2c4aa7a8c7734fa13da10723eb  
428  95d9bac431797e2cf8b7294a648ceec3bd50c84b  
429  76edb6d0bd7c47765d3a89032aa610f410506988  

[430 rows x 7 columns]

This is a really small dataset. So next step is to augment dataset with CoronaWhy dataset as well as Pubmed annotations.